### ISSUE TITLE
predict_and_sort

### ISSUE THREAD

**Issue by ZhigangSu** [ZhigangSu/Zhigang-Su#9]
import os
import torch
import torchvision.transforms as transforms
from PIL import Image
import shutil

import torch.nn as nn

class LeafDiseaseCNN(nn.Module):
    def __init__(self, num_classes):
        super(LeafDiseaseCNN, self).__init__()
        self.features = nn.Sequential(
            nn.Conv2d(3, 16, kernel_size=3, padding=1), nn.ReLU(), nn.MaxPool2d(2),
            nn.Conv2d(16, 32, kernel_size=3, padding=1), nn.ReLU(), nn.MaxPool2d(2)
        )
        self.classifier = nn.Sequential(
            nn.Flatten(),
            nn.Linear(32 * 32 * 32, 128), nn.ReLU(),
            nn.Linear(128, num_classes)
        )

    def forward(self, x):
        x = self.features(x)
        return self.classifier(x)

device = torch.device("cuda" if torch.cuda.is_available() else "cpu")

model = LeafDiseaseCNN(num_classes=2).to(device)
model.load_state_dict(torch.load("leaf_disease_model.pth", map_location=device))
model.eval()

transform = transforms.Compose([
    transforms.Resize((128, 128)),
    transforms.ToTensor()
])

input_folder = r"C:\Users\Ray\Documents\PlantDiseaseProject\unknown"
output_folder = r"C:\Users\Ray\Documents\PlantDiseaseProject\sorted"

healthy_dir = os.path.join(output_folder, "healthy")
diseased_dir = os.path.join(output_folder, "diseased")
os.makedirs(healthy_dir, exist_ok=True)
os.makedirs(diseased_dir, exist_ok=True)

for filename in os.listdir(input_folder):
    if filename.lower().endswith((".jpg", ".jpeg", ".png")):
        image_path = os.path.join(input_folder, filename)
        image = Image.open(image_path).convert("RGB")
        input_tensor = transform(image).unsqueeze(0).to(device)

        with torch.no_grad():
            output = model(input_tensor)
            _, predicted = torch.max(output, 1)

        if predicted.item() == 0:
            target_path = os.path.join(healthy_dir, filename)
            label = "healthy"
        else:
            target_path = os.path.join(diseased_dir, filename)
            label = "diseased"

        shutil.copy(image_path, target_path)
        print(f"{filename} → {label}")

print(" All unknown images have been classified and copied to 'sorted/' folder.")
